homeassistant/components/mqtt/climate.py

```diff
@@ -399,36 +389,53 @@ def config_schema():
 
     def _setup_from_config(self, config):
         """(Re)Setup the entity."""
+        self._attr_hvac_modes = self._config[CONF_MODE_LIST]
+        self._attr_min_temp = self._config[CONF_TEMP_MIN]
+        self._attr_max_temp = self._config[CONF_TEMP_MAX]
+        self._attr_precision = self._config.get(CONF_PRECISION, super().precision)
+        self._attr_fan_modes = self._config[CONF_FAN_MODE_LIST]
+        self._attr_swing_modes = self._config[CONF_SWING_MODE_LIST]
+        self._attr_target_temperature_step = self._config[CONF_TEMP_STEP]
+        self._attr_temperature_unit = self._config.get(
+            CONF_TEMPERATURE_UNIT, self.hass.config.units.temperature_unit
+        )
+
         self._topic = {key: config.get(key) for key in TOPIC_KEYS}
 
         # set to None in non-optimistic mode
-        self._target_temp = (
-            self._current_fan_mode
-        ) = self._current_operation = self._current_swing_mode = None
-        self._target_temp_low = None
-        self._target_temp_high = None
+        self._attr_target_temperature = (
+            self._attr_fan_mode
+        ) = self._attr_hvac_mode = self._attr_swing_mode = None
```

Comment:
It is clearer on independant lines:
```suggestion
        self._attr_target_temperature = None
        self._attr_fan_mode = None
        self._attr_hvac_mode = None
        self._attr_swing_mode = None
```